cpp/src/parquet/file_writer.cc

```diff
@@ -219,17 +219,17 @@ class RowGroupSerializer : public RowGroupWriter::Contents {
       closed_ = true;
       CheckRowsWritten();
 
-      for (size_t i = 0; i < column_writers_.size(); i++) {
-        if (column_writers_[i]) {
-          total_bytes_written_ += column_writers_[i]->Close();
+      // If ColumnWriter::Close thrown an exception, avoid
+      // have bad states in column_writers_.
```

Comment:
```suggestion
      // Avoid invalid state if ColumnWriter::Close() throws internally.
```

Reply:
If anything wrong happens here, should we let the FileSerializer know that it cannot accept any write any more?

Reply:
```c++
  void Close() override {
    if (!closed_) {
      closed_ = true;
      CheckRowsWritten();
 

      // ...

      // Ensures all columns have been written
      metadata_->set_num_rows(num_rows_);
      metadata_->Finish(total_bytes_written_, row_group_ordinal_);
    }
  }
```

I guess it will mark `closed_` first and throw exception. During stack unwind, the `RowGroupSerializer` will destruct. And during next time `Close` is called, it will do nothing